packages/card/src/vaadin-card.js

```diff
@@ -46,7 +299,64 @@ class Card extends ElementMixin(ThemableMixin(PolylitMixin(LitElement))) {
 
   /** @protected */
   render() {
-    return html`<slot></slot>`;
+    return html`
+      <div part="media">
+        <slot name="media"></slot>
+      </div>
+      <div part="header">
+        <slot name="header-prefix"></slot>
+        <slot name="header">
+          <slot name="title"></slot>
+          <slot name="subtitle"></slot>
+        </slot>
+        <slot name="header-suffix"></slot>
+      </div>
+      <div part="content">
+        <slot></slot>
+      </div>
+      <div part="footer">
+        <slot name="footer"></slot>
+      </div>
+    `;
+  }
+
+  /** @private */
+  _onSlotChange() {
+    // Chrome doesn't support `:host(:has())`, so we'll recreate that with custom attributes
+    this.toggleAttribute('has-media', this.querySelector(':scope > [slot="media"]'));
+    this.toggleAttribute('has-header', this.querySelector(':scope > [slot="header"]'));
+    this.toggleAttribute(
+      'has-title',
+      this.querySelector(':scope > [slot="title"]') && !this.querySelector(':scope > [slot="header"]'),
+    );
+    this.toggleAttribute(
+      'has-subtitle',
+      this.querySelector(':scope > [slot="subtitle"]') && !this.querySelector(':scope > [slot="header"]'),
+    );
+    this.toggleAttribute('has-header-prefix', this.querySelector(':scope > [slot="header-prefix"]'));
+    this.toggleAttribute('has-header-suffix', this.querySelector(':scope > [slot="header-suffix"]'));
+    this.toggleAttribute('has-content', this.querySelector(':scope > :not([slot])'));
+    this.toggleAttribute('has-footer', this.querySelector(':scope > [slot="footer"]'));
+  }
+
+  /** @protected */
+  connectedCallback() {
+    super.connectedCallback();
+
+    if (!CSS.supports('selector(:host(:has([slot])))')) {
+      if (!this.__boundSlotChangeListener) {
+        this.__boundSlotChangeListener = this._onSlotChange.bind(this);
+      }
+      this.shadowRoot.addEventListener('slotchange', this.__boundSlotChangeListener);
+    }
```

Comment:
It could be unexpected to have host attributes on some browsers and not on others. If we don't want to have the attributes in place for all browsers, an alternative could be to wrap the shadow root content inside an element with  `display: content` and add the attributes to it instead.

Reply:
Yeah, I think you’re right – this can cause confusion. I think an extra wrapper would be best, so that we don't end up exposing these attributes in the global scope at all, and make it impossible for themes and custom styling to rely on them (as they can use `vaadin-card:has()`).

An alternative might be to use more cryptic attribute names, which would disincentivize their use.

Reply:
With this uncertainty, I suppose it’s best to use the same implementation (apply the attributes) in all browsers.

Reply:
> An alternative might be to use more cryptic attribute names, which would disincentivize their use.

This is what I ended up doing. There are too many problems around `:host(:has())` and `:host:has()` that it seemed best to simplify the internal CSS at the expense of a little verbosity in the public facing DOM.